pkgs/development/libraries/memorymapping/default.nix

```
@@ -37,7 +37,9 @@ stdenv.mkDerivation {
     description = "fmemopen for Mac OS and iOS";
     license = licenses.asl20;
     maintainers = with maintainers; [ veprbl ];
-    platforms = platforms.unix;
+    # Uses BSD-style funopen() to implement glibc-style fmemopen().
+    # Add more BSDs if you need to.
+    platforms = platforms.darwin;
     broken = stdenv.isAarch64;
```

This is libc-specific so
```suggestion
    broken = stdenv.cc.isGNU && stdenv.isAarch64;
```

---

I suspect `isGNU` is really about `gcc/clang` compiler and not libc (it outgh to be named isGCC):

```
nix-repl> clangStdenv.cc.isGNU
false

nix-repl> stdenv.cc.isGNU
true

nix-repl> stdenv.hostPlatform.libc
"glibc"

nix-repl> clangStdenv.hostPlatform.libc
"glibc"
```

I guess what wee need is `clangStdenv.hostPlatform.libc NOT IN ("glibc", "musl")` or `clangStdenv.hostPlatform.libc IN (${BSD_VARIANTS})`.

---

Fair point. So you imply that musl is also not supported? If that's the case, your original change makes more sense.

---

I see it is: http://git.musl-libc.org/cgit/musl/tree/include/stdio.h?id=f368d9fd26ae002fe2fce20add4cb2b806f48972#n58